hrjobcontract/CRM/Hrjobcontract/Test/BaseHeadlessTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+
+abstract class CRM_Hrjobcontract_Test_BaseHeadlessTest extends PHPUnit_Framework_TestCase
+  implements HeadlessInterface, TransactionalInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+               ->install('uk.co.compucorp.civicrm.hrcore')
+               ->install('uk.co.compucorp.civicrm.hrleaveandabsences')
+               ->installMe(__DIR__)
+               ->apply();
+  }
+
+  /**
+   * wrap api functions.
+   * so we can ensure they succeed & throw exceptions without litterering the test with checks
+   *
+   * @param string $entity
+   * @param string $action
+   * @param array $params
+   * @param mixed $checkAgainst
+   *   Optional value to check result against, implemented for getvalue,.
+   *   getcount, getsingle. Note that for getvalue the type is checked rather than the value
+   *   for getsingle the array is compared against an array passed in - the id is not compared (for
+   *   better or worse )
+   *
+   * @return array|int
+   */
+  public function callAPISuccess($entity, $action, $params, $checkAgainst = NULL) {
+    $params = array_merge(['debug' => 1], $params);
+    switch (strtolower($action)) {
+      case 'getvalue':
+        return $this->callAPISuccessGetValue($entity, $params, $checkAgainst);
```

Comment:
If these functions are public methods that can be called directly, Why do we have them in the `callAPISuccess` function also ?  I initially thought they were private methods and would be exposed via `callAPISuccess`